accounts/abi/bind/bind.go

```diff
@@ -74,12 +74,31 @@ func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]
 			transacts = make(map[string]*tmplMethod)
 			events    = make(map[string]*tmplEvent)
 			structs   = make(map[string]*tmplStruct)
+
+			// identifiers are used to detect duplicated identifier of function
+			// and event. For all calls, transacts and events, abigen will generate
+			// corresponding bindings. However we have to ensure there is no
+			// identifier coliision in the bindings of these categories.
+			//
+			// The key of identifiers map is the concat of function name and function type.
+			identifiers = make(map[string]bool)
 		)
 		for _, original := range evmABI.Methods {
 			// Normalize the method for capital cases and non-anonymous inputs/outputs
 			normalized := original
-			normalized.Name = methodNormalizer[lang](original.Name)
-
+			// Ensure there is no duplicated identifier
+			normalizedName := methodNormalizer[lang](alias(aliases, original.Name))
+			id := normalizedName
+			if original.Const {
+				id += "call"
```

Comment:
instead of having string concatenations, how about using three different identifier maps, `transactIdentifiers`, `callIdentifiers` and `eventIdentifiers` ?